### PR TITLE
Responsive Loader

### PR DESCRIPTION
**Pull request by hsab**
Adding dependency [responsive-loader](https://github.com/herrstucki/responsive-loader) to support automatic optimization of images. 

Modification to:

- Image component 09f2666c2bfd9ec2cb3f5eef3b764227f74bdf23
  - ```
    -     <Image src={GrahamImage1} alt=""  />
    +     <Image img={GrahamImage1} alt=""  />
    ```
  - https://github.com/blerchin/dma-20-mfa-show/blob/5e310f075f9c561c5e01a25a6ae176d270d88f9f/src/work/ErinCooney.js#L38-L41

  - https://github.com/blerchin/dma-20-mfa-show/blob/fc77442347148919bc3b018a7e0592dca9208949/src/work/ErinCooney.js#L35-L38

- `webpack.config.js` https://github.com/blerchin/dma-20-mfa-show/commit/5e310f075f9c561c5e01a25a6ae176d270d88f9f#diff-a58d55bdb5770c78ad512f8e91f8d051
  - Define appropriate sizes/formats in `webpack.config.js` https://github.com/blerchin/dma-20-mfa-show/blob/5e310f075f9c561c5e01a25a6ae176d270d88f9f/config/webpack.config.js#L353

Rebased to b3dd9aa9d948fc3e1de57f2623fc8bc40012a4b6 to pass netlify.
Example page: [Graham](https://deploy-preview-24--kind-tesla-a088c1.netlify.app/graham-akins)

### CODE REVIEW

**Comment by blerchin** [comment]
Let's wait till #23 is merged since this includes a lot of those commits. but otherwise LGTM

**Comment by hsab** [comment]
No worries. I just made another commit https://github.com/blerchin/dma-20-mfa-show/commit/8bda30893c4039d08cb7ed24c07fad0d6e2f712f. Noticed some of the pngs are larger than the original files. 

It had to do with sharp. Going over the [documentation](https://sharp.pixelplumbing.com/api-output#png) I realized they have support for `libimagequant` through `libvips`. But the default npm installed libvips is not compiled against this dependency. Anyway managed to install it system wide and was picked up by sharp. File size of 2.7mb is reduced to 860kb. 

But I was wondering how this plays out on the server? Or is there an option to build a static site from our app for deployment/dist?

**Comment by hsab** [comment]
Just tested on netlify, and seems like their servers have libvips compiled against libimagequant installed globally. So it's picked up on the deploy-previews and functions as expected. 

While I'm at it. I'm gonna try and figure out lazy loading.

**Comment by blerchin** [comment]
> But I was wondering how this plays out on the server? Or is there an option to build a static site from our app for deployment/dist?

Hmm yeah I hadn't thought about that. Netlify publish the [docker images they use internally](https://github.com/netlify/build-image), so we could set up Travis to run the same image and deploy the dist files as a release on Github. Then I guess someone has to copy those files to a DMA server lol.

Edit: I think I answered a different question than you were asking, but  it was good to think ahead to when we actually have to deploy and host this thing. Anyway that repo is a good source if you want to know exactly how Netlify sets up their VMs.

**Comment by brysonian** [comment]
I *think* it can be done via a github action, build and then scp to the dma server. But that'll take some fiddling. Or since this isn't really going to be updated once it is done, I can just build it locally and upload it manually.

File: src/Components/Image/index.js [hunk 1]
```diff
@@ -1,66 +1,69 @@
...
+          />
+        </div>
+      }
+      {caption && <div className={cap}>{caption}</div>}
+    </div>
+  );
+};
 
 Image.defaultProps = {
+<<<<<<< HEAD
```
Comment (blerchin) [inline src/Components/Image/index.js]: looks like the merge got committed here

**Review by blerchin — CHANGES_REQUESTED** [review]

File: src/Components/Image/index.js (hunk 1 above)
Comment (hsab) [reply src/Components/Image/index.js]: yea missed that one

Comment (hsab) [reply src/Components/Image/index.js]: this now also includes [react-lazy-load-image-component](https://github.com/Aljullu/react-lazy-load-image-component) dependency

**Review by hsab — COMMENTED** [review]
fixed in https://github.com/blerchin/dma-20-mfa-show/commit/0b23189fa14ec38e5e5cfc4f54e1e7d5d1b1b355

**Review by blerchin — APPROVED** [review]
👍 LGTM

**Comment by hsab** [comment]
> I _think_ it can be done via a github action, build and then scp to the dma server. But that'll take some fiddling. Or since this isn't really going to be updated once it is done, I can just build it locally and upload it manually.

Let's give it a 10-15 days post-opening grace period before archiving on DMA? I'm sure some folks will decide to change wording or make minor changes. I'll probably be in that boat.

**Pull request merged**